WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -62,6 +123,118 @@ class BattleView extends React.Component {
         }
     }
 
+    // helper function to set property states
+    setNewStateAttributes(state, property, attributes) {
+        if((state.visible !== undefined) && (attributes.visible !== undefined))
+            state.visible = attributes.visible;
+        if((state.opacity !== undefined) && (attributes.opacity !== undefined))
+            state.opacity = attributes.opacity;
+        if((state.scale !== undefined) && (attributes.scale !== undefined))
+            state.scale = attributes.scale;
+        if((state.translateY !== undefined) && (attributes.translateY !== undefined))
+            state.translateY = attributes.translateY;
+        this.setState({
+           [property]: state
+        });
+    }
+
+    // prototype function, runs if we click on mini enemy cards
+    changeCardsEnemyOrder = () => {
+        this.setState({
+            cardsEnemyOrder: [3, 2, 1, 5, 4]
+            // this means: first card on third place, second on second, third on first and so on
+        });
+    }
+
+    // prototype function, runs if we click on mini user cards
+    changeCardsUserOrder = () => {
+        this.setState({
+            cardsUserOrder: [5, 4, 2, 3, 1]
+            // this means: first card on fifth place, second on fourth, third on second and so on
+        });
+    }
+
+    // get proper mini cards to DOM
+    getMiniCards = (enemy) => {
+        return (
+            [...Array(5)].map(
+                (e,i) => {
+                    return (
+                        <MiniCardView key={enemy ? `enemyCard-${i}` : `userCard-${i}`}

```

Comment:
Nie możemy zaifować przed returnem i zwrócić odpowiednio kartę dla usera / enemy? Te ify w środku jsx się ciężko czyta

Reply:
Możemy, ja po prostu już się przyzwyczaiłem do tego zapisu ? :, i czyta mi się tak lepiej, bo mniej linijek

Reply:
a może o to Ci chodziło
![image](https://user-images.githubusercontent.com/51495505/142175525-02bd922d-56f9-49c2-aef3-82ffa52e2d88.png)
ale nie mogę tak, bo nie mam dostępu do indexu

Reply:
drugie co mi przychodzi do głowy to zaifować cały ten .map, ale wolałbym tego nie robić bo wtedy będzie jeszcze mniej czytelne moim zdaniem, dużo linijek

Reply:
To co proponujesz wygląda git